packages/flutter/lib/src/material/input.dart

```diff
@@ -95,49 +128,124 @@ class InputState extends ScrollableState<Input> {
       config.onSubmitted(_value);
   }
 
-  Widget buildContent(BuildContext context) {
+  Widget build(BuildContext context) {
     assert(debugCheckHasMaterial(context));
     ThemeData themeData = Theme.of(context);
     bool focused = Focus.at(context, autofocus: config.autofocus);
 
-    if (focused && !_keyboardHandle.attached) {
-      _keyboardHandle = keyboard.show(_editableString.stub, config.keyboardType);
-      _keyboardHandle.setText(_editableString.text);
-      _keyboardHandle.setSelection(_editableString.selection.start,
-                                   _editableString.selection.end);
-    } else if (!focused && _keyboardHandle.attached) {
-      _keyboardHandle.release();
-    }
+    _attachOrDetachKeyboard(focused);
 
-    TextStyle textStyle = themeData.text.subhead;
-    List<Widget> textChildren = <Widget>[];
+    TextStyle textStyle = config.style ?? themeData.text.subhead;
+    Color focusHighlightColor = themeData.accentColor;
+    if (themeData.primarySwatch != null)
+      focusHighlightColor = focused ? themeData.primarySwatch[400] : themeData.hintColor;
+    double topPadding = config.isDense ? 12.0 : 16.0;
```

Comment:
Yeah, these values just come straight from the spec.  They aren't re-used anywhere as far as I can tell.